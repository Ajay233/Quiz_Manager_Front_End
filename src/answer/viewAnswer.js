import React from 'react';
import { connect } from 'react-redux'
import UpdateAnswerForm from '../forms/updateAnswer'
import NewAnswerForm from '../forms/newAnswer'
import Notification from '../notifications/notifications'
import history from '../history'

export class AnswerView extends React.Component {

  componentDidMount(){
    document.documentElement.scrollTop = 0;
  }

  renderForm = () => {
    return history.location.pathname === "/newAnswer" ? <NewAnswerForm /> : <UpdateAnswerForm />;
  }

  render(){
    return(
      <div className="componentContainer">
        <div>
          <Notification />
        </div>
        {this.renderForm()}
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    currentAnswer: state.currentAnswer
  }
}

export default connect(mapStateToProps)(AnswerView)
