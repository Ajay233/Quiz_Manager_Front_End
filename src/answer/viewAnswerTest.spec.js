import React from 'react'
import { Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import { AnswerView, mapStateToProps  } from './viewAnswer'
import configureStore from 'redux-mock-store'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import history from '../history'

const mockStore = configureStore({})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: { id: 1, jwt: "jwt" },
      currentAnswer: { id: 1 }
    }

    const componentState = mapStateToProps(appState)
    expect(appState).toEqual(componentState)
  })
})

describe("AnswerView", () => {
  it("should render the NewAnswerForm", () => {
    const store = mockStore({
      notificationData: {
        message: "",
        type: "",
        show: false,
        timed: true
      }
    })
    history.push("/newAnswer")
    const component = renderer.create(
      <Provider store={store}>
      <Router history={history}>
        <AnswerView />
      </Router>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it("should render the UpdateAnswerForm", () => {
    const store = mockStore({
      notificationData: {
        message: "",
        type: "",
        show: false,
        timed: true
      },
      currentAnswer: {
        id: 1,
        answerNumber: 1,
        description: "testDescription",
        correctAnswer: true
      }
    })
    history.push("/updateAnswer")
    const component = renderer.create(
      <Provider store={store}>
      <Router history={history}>
        <AnswerView />
      </Router>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })
})
