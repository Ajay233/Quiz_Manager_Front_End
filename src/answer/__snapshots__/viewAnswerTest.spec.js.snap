// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerView should render the NewAnswerForm 1`] = `
<div
  className="componentContainer"
>
  <div />
  <div
    className="componentContainer"
  >
    <div
      className="title-large-spaced"
    >
      Create an Answer
    </div>
    <form
      className="form-centered"
      onSubmit={[Function]}
    >
      <div>
        <label>
          Answer number:
        </label>
        <input
          className="inputBox"
          name="number"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
      <div>
        <label>
          Answer description
        </label>
        <textarea
          className="inputBox"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
      <div>
        <label>
          Correct? Yes/No:
        </label>
        <div>
          <select
            className="inputBox select-medium"
            name="correct"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Select an option
            </option>
            <option
              value={true}
            >
              Yes
            </option>
            <option
              value={false}
            >
              No
            </option>
          </select>
        </div>
      </div>
      <div
        className="buttonGroup"
      >
        <button
          className="submit"
        >
          Save
        </button>
        <a
          className="cancel"
          href="/editQuestion"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </form>
  </div>
</div>
`;

exports[`AnswerView should render the UpdateAnswerForm 1`] = `
<div
  className="componentContainer"
>
  <div />
  <div>
    <div
      className="title-large-spaced"
    >
      Edit Answer 1
    </div>
    <form
      className="form-centered"
      onSubmit={[Function]}
    >
      <div>
        <label>
          Answer number:
        </label>
        <input
          className="inputBox"
          name="number"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
      <div>
        <label>
          Answer description
        </label>
        <textarea
          className="inputBox"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
      <div>
        <label>
          Correct? Yes/No:
        </label>
        <div>
          <select
            className="inputBox select-medium"
            name="correct"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option
              value={true}
            >
              Yes
            </option>
            <option
              value={false}
            >
              No
            </option>
          </select>
        </div>
      </div>
      <div
        className="buttonGroup"
      >
        <button
          className="submit"
        >
          Save
        </button>
        <a
          className="cancel"
          href="/editQuestion"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </form>
  </div>
</div>
`;
